
#######################
Kegg metabolic pathways
#######################

How to read the table ?
-----------------------

***Label**: Label of the protein. If you click on the label, you access to the Gene annotation window
***Region**: '+' indicate the genes presence in the region of interest and '-' indicate abscence
***Gene**: Gene name of the protein
***EC number**: EC number associated with the protein
***Product**: Product description of the protein
***Evidence**: Status of the annotation: "validated" indicate the gene is manual annotated and "automatic" indicate the gene annotation come from MicroScope system annotation 
***Enzyme Description**:
***KO**: ID number of KEGG Ortholog entry
***KO Definition**: Functional description of KO
***KO EC number **: EC number associated with the KO

In the table (3), each line describes a gene related to an enzymatic reaction or a KO of this pathway. EC numbers (enzymatic functions) and the predictions of KofamScan are useful to construct these links. The « region » column indicates the genes presence/absence in the region of interest.

On the KEGG maps (4), reactions matching genome annotations are highlighted in green and reaction matching region annotations are highlighted in yellow. More details are available by clicking on items of the map and. The Reload button allows the user to come back in this his exploration work.
